org.sonarlint.eclipse.core/src/org/sonarlint/eclipse/core/internal/preferences/SonarLintProjectConfigurationManager.java

```diff
@@ -54,15 +57,22 @@ public class SonarLintProjectConfigurationManager {
 
   private static final Set<String> BINDING_RELATED_PROPERTIES = Set.of(P_PROJECT_KEY, P_CONNECTION_ID, P_BINDING_SUGGESTIONS_DISABLED_KEY);
 
+  private static final Map<ISonarLintProject, IPreferenceChangeListener> projectBindingPropertiesListener = new HashMap<>();
+
   public static void registerPreferenceChangeListenerForBindingProperties(ISonarLintProject project, Consumer<ISonarLintProject> listener) {
+    projectBindingPropertiesListener.put(project, event -> {
+      if (BINDING_RELATED_PROPERTIES.contains(event.getKey())) {
+        listener.accept(project);
+      }
+    });
+
     ofNullable(project.getScopeContext().getNode(SonarLintCorePlugin.PLUGIN_ID))
-      .ifPresent(node -> {
-        node.addPreferenceChangeListener(event -> {
-          if (BINDING_RELATED_PROPERTIES.contains(event.getKey())) {
-            listener.accept(project);
-          }
-        });
-      });
+      .ifPresent(node -> node.addPreferenceChangeListener(projectBindingPropertiesListener.get(project)));
+  }
+
+  public static void removePreferenceChangeListenerForBindingProperties(ISonarLintProject project) {
+    ofNullable(project.getScopeContext().getNode(SonarLintCorePlugin.PLUGIN_ID))
+      .ifPresent(node -> node.removePreferenceChangeListener(projectBindingPropertiesListener.remove(project)));
```

Comment:
What if `remove` returns null? Should we check that? Or `removePreferenceChangeListener` can accept Nullable values? Maybe this cannot happen, but just to be on the safe-side we could check